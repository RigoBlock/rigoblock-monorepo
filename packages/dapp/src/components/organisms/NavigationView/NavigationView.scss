@import "../../_settings/variables.scss";

.navigation-view {
  > *:not(:first-child) {
    padding-top: $dist-xs;
  }
}
