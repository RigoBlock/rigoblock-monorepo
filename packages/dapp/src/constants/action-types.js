import keyMirror from 'keymirror'

export const actionTypes = keyMirror({
  // Global actions
  GLOBAL_INIT: null,
  // Blockchain actions
  BLOCKCHAIN_INIT: null,
  BLOCKCHAIN_ERROR: null,
  LOGGED_IN: null,
  LOGGED_OUT: null
})
