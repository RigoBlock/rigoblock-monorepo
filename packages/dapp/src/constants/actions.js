export default {
  persist: 'persist',
  form: '@@redux-form'
}
