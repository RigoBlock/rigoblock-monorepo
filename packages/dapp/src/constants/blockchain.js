export const VAULT = 'vault'
export const DRAGO = 'drago'
