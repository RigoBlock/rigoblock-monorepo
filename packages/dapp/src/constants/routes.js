export const DASHBOARD = '/'
export const PREFERENCES = '/preferences'
export const HELP = '/help'
export const LOGIN = '/login'
export const VAULTS = '/vaults'
