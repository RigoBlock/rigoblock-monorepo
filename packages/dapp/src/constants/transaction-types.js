export const CREATE_VAULT = 'Create Vault'
