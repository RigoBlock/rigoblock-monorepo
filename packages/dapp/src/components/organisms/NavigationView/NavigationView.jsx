import './NavigationView.scss'
import * as ROUTES from '../../../constants/routes'
import List from '../List'
import PropTypes from 'prop-types'
import React from 'react'
import ViewLink from '../../molecules/ViewLink'

const NavigationView = () => {
  const navLinks = [
    {
      icon: 'dashboard',
      link: ROUTES.DASHBOARD
    },
    {
      icon: 'lock',
      link: ROUTES.VAULTS
    }
  ]
  return (
    <div className="navigation-view">
      <List Component={ViewLink} data={navLinks} />
    </div>
  )
}

NavigationView.propTypes = {
  accounts: PropTypes.object.isRequired
}

export default NavigationView
