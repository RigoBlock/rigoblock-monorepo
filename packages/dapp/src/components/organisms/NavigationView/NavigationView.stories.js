import '../../_settings/_base.scss'
import * as ROUTES from '../../../constants/routes'
import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import { selectV2, withKnobs } from '@storybook/addon-knobs/react'
import { storiesOf } from '@storybook/react'
import NavigationView from './NavigationView'
import React from 'react'

const mockStore = {
  getState: () => ({
    routing: {
      location: {
        pathname: selectV2('url', ROUTES, ROUTES.DASHBOARD)
      }
    }
  }),
  dispatch: () => null,
  subscribe: () => null
}

storiesOf('Organisms/NavigationView', module)
  .addDecorator(withKnobs)
  .addDecorator(story => <MemoryRouter>{story()}</MemoryRouter>)
  .addDecorator(story => <Provider store={mockStore}>{story()}</Provider>)
  .add('default', () => <NavigationView />)
