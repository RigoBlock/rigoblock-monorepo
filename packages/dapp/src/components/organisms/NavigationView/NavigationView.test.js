import * as ROUTES from '../../../constants/routes'
import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import { mount, shallow } from 'enzyme'
import NavigationView from './NavigationView.jsx'
import React from 'react'
import toJson from 'enzyme-to-json'

const mockStore = {
  getState: () => ({
    routing: {
      location: {
        pathname: ROUTES.DRAGOS
      }
    }
  }),
  dispatch: jest.fn(),
  subscribe: () => null
}

const wrapper = mount(
  <Provider store={mockStore}>
    <MemoryRouter>
      <NavigationView />
    </MemoryRouter>
  </Provider>
)

const shallowWrapper = shallow(<NavigationView store={mockStore} />).dive()

describe('NavigationView component', () => {
  it('renders correctly', () => {
    expect(toJson(shallowWrapper)).toMatchSnapshot()
  })

  it(`adds active class to drago link if we are in ${
    ROUTES.DRAGOS
  } route`, () => {
    const dragoLink = wrapper.find(`a[href='${ROUTES.DRAGOS}']`)
    expect(dragoLink.hasClass('active')).toBe(true)
  })
})
