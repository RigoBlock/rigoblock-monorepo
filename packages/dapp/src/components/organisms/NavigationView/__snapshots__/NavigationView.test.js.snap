// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationView component renders correctly 1`] = `
<div>
  <List
    Component={[Function]}
    Divider={null}
    className="navigation-view"
    data={
      Array [
        Object {
          "className": "",
          "icon": "dashboard",
          "link": Object {
            "text": "Dashboard",
            "to": "/",
          },
        },
        Object {
          "className": "",
          "icon": "account_balance_wallet",
          "link": Object {
            "text": "Dragos",
            "to": "/dragos",
          },
        },
        Object {
          "className": "active",
          "icon": "lock",
          "link": Object {
            "text": "Vaults",
            "to": "/vaults",
          },
        },
      ]
    }
  />
</div>
`;
